studio/src/aem/aem-fragments.js

```diff
@@ -179,9 +178,13 @@ class AemFragments extends LitElement {
         const fragment = await this.#aem.sites.cf.fragments.getById(
             this.searchText,
         );
-        this.#searchResult = [fragment];
+        if (!fragment || fragment.path.indexOf(getDamPath(this.path)) < 0) {
+            this.#searchResult = [];
+        } else {
+            this.#searchResult = [fragment];
+            this.addToCache([fragment]);
```

Comment:
this needs an await, maybe you could add it, otherwise it should come with Ilyases changes